### ISSUE TITLE
Empty literal string in a list of strings not parsed correctly

### ISSUE THREAD

**Issue by iquito** [vimeo/psalm#2448]
If you define a list of allowed literal strings for a class property and include "" as the first one (empty string), Psalm incorrectly parses the whole expression:

```php
class User
{
  /**
   * @var string
   * @psalm-var ""|"admin"|"fun"
   */
  private $type = '';
}
```

Psalm reports `ERROR: InvalidPropertyAssignmentValue - 9:19 - $this->type with declared type 'string("|)' cannot be assigned type 'string()'`.

If you switch it around, like this:

```php
class User
{
  /**
   * @var string
   * @psalm-var "admin"|"fun"|""
   */
  private $type = '';
}
```

Then it works, but one might want to have an empty string as the first option, as it might be the default one. See error on psalm.dev: https://psalm.dev/r/cf4bdecf70

This issue also happens with return types, so I suspect Psalm does not expect a literal empty string as the first one in a list wherever types are defined. If you only define an empty string it works:

```php
class User
{
  /**
   * @var string
   * @psalm-var ""
   */
  private $type = '';
}
```

This reports no issues, only when a `|` comes into play the error appears.

**Comment by muglug** [comment]
Another workaround is to put those values in constants, but putting the empty string in a constant is also a little strange.

**Comment by muglug** [comment]
Fixed in 79f3a5d - off-by-one errors FTW
